core/commands/ping.go

```diff
@@ -0,0 +1,129 @@
+package commands
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	cmds "github.com/jbenet/go-ipfs/commands"
+	peer "github.com/jbenet/go-ipfs/p2p/peer"
+	u "github.com/jbenet/go-ipfs/util"
+
+	context "github.com/jbenet/go-ipfs/Godeps/_workspace/src/code.google.com/p/go.net/context"
+)
+
+const kPingTimeout = 10 * time.Second
+
+type PingResult struct {
+	Success bool
+	Time    time.Duration
+}
+
+var PingCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "send echo request packets to IPFS hosts",
+		Synopsis: `
+ipfs ping <peer.ID> - Send pings to a peer using the routing system to discover its address
+`,
+		ShortDescription: `
+ipfs ping is a tool to find a node (in the routing system),
+send pings, wait for pongs, and print out round-trip latency information.
+`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("count", false, true, "Number of pings to perform"),
```

Comment:
this should be a `cmds.Option`, and you can have a StringArg for the peer ID:

```Go
Arguments: []cmds.Argument{
  cmds.StringArg("peer", true, true, "id of peer to ping"),
},
Options: []cmds.Option{
  cmds.IntOption("count", "c", "number of pings to perform"),
},
```

see `ipfs ls` or other commands like it